src/storage/leveldb/history_index.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+import { Level } from 'level';
+import { AbstractSublevel } from 'abstract-level';
+import { IKVHistoryIndex, IHistoryTx, HistoryIndexValidateResponse } from '../../types';
+import { errorCodeOrNull, KEY_NOT_FOUND_CODE } from './errors';
+
+export const HISTORY_PREFIX = 'history';
+export const TS_HISTORY_PREFIX = 'ts_history';
+
+
+function _ts_key(tx: Pick<IHistoryTx, 'timestamp'|'tx_id'>): string {
+  const buf = Buffer.alloc(4);
+  buf.writeUint32BE(tx.timestamp);
+  return `${buf.toString('hex')}:${tx.tx_id}`;
+}
+
+export default class LevelHistoryIndex implements IKVHistoryIndex {
+  dbpath: string;
+  historyDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, IHistoryTx>;
+  tsHistoryDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, IHistoryTx>;
+  size: number;
+  indexVersion: string = '0.0.1';
+
+  constructor(dbpath: string) {
+    this.dbpath = path.join(dbpath, 'history');
+    const db = new Level(this.dbpath);
+    this.historyDB = db.sublevel<string, IHistoryTx>(HISTORY_PREFIX, { valueEncoding: 'json' });
+    this.tsHistoryDB = db.sublevel<string, IHistoryTx>(TS_HISTORY_PREFIX, { valueEncoding: 'json' });
+    this.size = 0;
```

Comment:
Why didn't you continue with this external count?

Reply:
There were a couple issues during the initial implementation so i went with the simpler approach but we can still change this since its internal to the index.

Reply:
Added a counter on the database with validation on both address and history indexes